### PR TITLE
9235 collapse search filters

### PR DESCRIPTION
**Pull request by rhianwatkins1**
# 9235 Collapsable Search Filters

[TP9235](https://moneyadviceservice.tpondemand.com/entity/9235-collapsable-search-filters-mobile) 

This PR makes the filters collapsable on mobile

##Screenshots
**Before**
screen-shot-2018-06-20-at-09 37 02

**After**
screen-shot-2018-06-20-at-09 36 48
screen-shot-2018-06-20-at-11 18 58
screen-shot-2018-06-20-at-11 19 07

### CODE REVIEW

**Comment by davidtrussler** [comment]
👍 - looks good!

File: app/assets/javascripts/components/Filters.js [hunk 1]
```diff
@@ -43,5 +55,53 @@ define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
...
+    }
+    self.windowSize = $(window).width();
+  } 
+
+  /**
+  * Is Mobile
+  *   was previously desktop, or page just loaded, so rest for mobile
+  */
+  Filter.prototype._toggleMobile = function(self, isLoad) {
+    if ( self.windowSize >this.breakpoint || isLoad ) {
```
Comment (tomas-stefano) [inline app/assets/javascripts/components/Filters.js]: I would format the line with less spaces around the if and spaces for the comparison like.`if (self.windowSize > this.breakpoint || isLoad)`. But that's just my taste.

File: app/assets/javascripts/components/Filters.js [hunk 2]
```diff
@@ -43,5 +55,53 @@ define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
     this.$form.submit();
   }
 
+  /**
+  * Assign/Remove listner on toggle
+  */
+  Filter.prototype._toggleFilters = function(self, isLoad) {  
+    if ( $(window).width() < this.breakpoint ) {
```
Comment (tomas-stefano) [inline app/assets/javascripts/components/Filters.js]: Can this verification be a function? `$(window).width() < this.breakpoint`

File: app/assets/javascripts/components/Filters.js [hunk 3]
```diff
@@ -43,5 +55,53 @@ define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
...
+        self._toggleAnimate();
+      })
+    }
+  }
+  /**
+  * Is Desktop
+  *   was previously mobile, or page just loaded, so rest for desktop
+  */
+  Filter.prototype._toggleDesktop = function(self, isLoad) {
+    if ( self.windowSize < this.breakpoint || isLoad ) {
```
Comment (tomas-stefano) [inline app/assets/javascripts/components/Filters.js]: If I understood correctly, this verification is duplicated. Can this verification be a function? Wdyt?

File: app/assets/javascripts/components/Filters.js (hunk 1 above)
Comment (rhianwatkins1) [reply app/assets/javascripts/components/Filters.js]: done

File: app/assets/javascripts/components/Filters.js (hunk 3 above)
Comment (rhianwatkins1) [reply app/assets/javascripts/components/Filters.js]: There is a similar if statement on line 75 but they are not the same

Comment (rhianwatkins1) [reply app/assets/javascripts/components/Filters.js]: Scrap that, I've put the first part of the if statement in a function

**Pull request merged**